clients/privacy-center/app/server-environment.ts

```diff
@@ -0,0 +1,206 @@
+/**
+ * Utility functions and logic that is designed to exclusively run server-side to configure the environment for the app, e.g.:
+ * 1) Securely loading ENV variables for the client to use
+ * 2) Fetching configuration files (config.json, config.css) to inject into the client
+ * 3) etc.
+ * 
+ * During server-side rendering, call loadPrivacyCenterEnvironment() to initialize the environment values for the App.
+ * 
+ * On the client-side, call the usePrivacyCenterEnvironment() hook in code to safely access the environment values at runtime.
+ */
+import { URL } from "url";
+
+import { Config } from "~/types/config";
+
+/************************
+ * SERVER-SIDE functions
+ ************************/
+
+/**
+ * Settings that can be controlled using ENV vars on the server.
+ * 
+ * Any of these can be set by adding the prefix "FIDES_PRIVACY_CENTER__", e.g.
+ * 
+ * FIDES_PRIVACY_CENTER__FIDES_API_URL=https://fides.example.com/api/v1
+ */
+export interface PrivacyCenterSettings {
+  FIDES_API_URL: string; // e.g. http://localhost:8080/api/v1
+  CONFIG_JSON_URL: string; // e.g. file:///app/config/config.json
+  CONFIG_CSS_URL: string; // e.g. file:///app/config/config.css
+}
+
+/**
+ * Environment that is generated server-side and provided to the client
+ */
+export interface PrivacyCenterEnvironment {
+  fidesApiUrl: string;
+  config?: Config;
+  styles?: string;
+}
+
+/**
+ * Loads all the ENV variable settings, configuration files, etc. to initialize the environment
+ */
+let _environment: PrivacyCenterEnvironment;
+export const loadPrivacyCenterEnvironment = async () => {
+  if (typeof window !== "undefined") {
+    throw new Error("Unexpected error, cannot load server environment from client code!");
+  }
+  console.log("Load Privacy Center environment for session..."); // TODO: version number
```

Comment:
TODO: I'd like to log a version number on the server-side, but not a hard requirement